### DISCUSSION TITLE
USB Host

### DISCUSSION THREAD

**Discussion by daltronbeats** [eh2k/squares-and-circles#78] [Q&A]
I noticed a mention of your interest in USB host in a previous comment. Was this ever implemented? I'd love to make a dedicated midi expander with seperate pots for each machine.

(Also looking forward to details on the FV1 implementation! I'm going to have to start making my 4th squares and circles soon 😁)

**Comment by eh2k** [comment]
@daltronbeats,

i'm not sure i understand the question correctly. By USB host, do you mean USB-midi support?

Regarding Midi Expander.
My latest findings are that the following user interface would be quite practical:

![](https://user-images.githubusercontent.com/11786292/257056387-a23870fa-72df-43c8-93cf-e6598c33a1d6.svg)

Regarding FV1 .... more info when I am ready.

**Comment by daltronbeats** [comment]
This looks great! Are the BD1, BD2 etc extra gate inputs?for additional engines beyond 1-4? 
Would be happy to beta test any new hardware when you are ready! That is my expertise..

Sorry I wasn't clear. The teensy 4.0 has 2 pins on the rear for USB host support. If I connect a class compliant usb midi controller to those would it work? (I'm already using the teensy main USB as a midi interface for triggering notes from a PC)

I understand I could send MIDI CC from the PC but it would be nice to have a small midi controller module next to my squares and circles module OR be able to connect a USB midi controller when I am away from the PC

**Comment by eh2k** [comment]
I haven't done anything with the USB host on the Teensy yet. However, I have experimented with USB host on the DaisySM.

In general, I would probably prefer the serial/midi interface for an external controller. I had already thought about using the pins on the back for the Midi-Expander (you would have to solder something like this https://mou.sr/45eRDG8 to the teensy).

Regarding the second prototype:
The BD1, BD2 would actually just be buttons (with LEDs, like https://www.thonk.co.uk/shop/illuminated-tact-switches/). They are primarily used to select the track or to visualize the selection or triggs (these could also be named differently). The other two buttons (left + right) would be for navigation between the paramter page and io page - or extra pages like mixing. 1-8 would be CV inputs that also serve as triggers. A,B could be audio inputs, for example. C,D are the audio outputs. The current firmware concept would actually fit well with this. With only two audio inputs and mixing, 6 tracks would also be conceivable. At the moment there are 4 tracks/engines, as it fits well with the 4 DAC outputs.

**Reply by vestedaf** [reply]
Hi there,

I've also found a desire to hook up a small USB midi keyboard to play the VAx6 engine. It doesn't look like anything has been done to allow for USB Host duties, is that correct? Could this be revisited? I think this would be a really useful feature, since the T4.0 should be able to support this.

Thanks for considering!
